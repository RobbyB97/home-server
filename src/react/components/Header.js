import React from 'react'
import {NavLink} from 'react-router-dom'
import {connect} from 'react-redux'

import toggleNav from '../actions/navigation'
import StyleGuide from './StyleGuide'
import NavigationMenu from './NavigationMenu'

const Header = ({nav, dispatch}) => (
  <header id="head">
    <div id="head-logo">
      <a href='#'>
        <img src='/assets/images/favicon.png' alt='Home'/> 
      </a>
    </div>
    <button id="nav-button" onClick={() => {
      dispatch(toggleNav())
    }}>
      {!nav ? '☰':'✕'}
    </button>
    <h1>{!!nav}</h1>
  </header>
) 

const mapStateToProps = (state) => {
  return {
    nav: state.navigation.nav
  }
}

export default connect(mapStateToProps)(Header)