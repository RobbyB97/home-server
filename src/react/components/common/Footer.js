import React from 'react'


export class Footer extends React.Component {
    
    render() {
        return (
            <div id="footer">
                <div class="footer__logos">

                </div>
                <div class="footer__copyright">
                    
                </div>
            </div>
        )
    }
}


export default Footer