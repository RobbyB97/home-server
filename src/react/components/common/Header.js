import React from 'react'
import {NavLink} from 'react-router-dom'
import {connect} from 'react-redux'

import {toggleNav, toggleNavOff} from '../../actions/navigation'
import {setTitle} from '../../actions/title'
import {homeOn} from '../../actions/home'


const Header = ({nav, dispatch}) => (
	<header id="header">
		<div className="header__logo">
	  		<NavLink to="/react/" onClick={() => {
				dispatch(setTitle(''))
				dispatch(toggleNavOff())
				dispatch(homeOn())
	  		}}>
			<img src='/assets/images/favicon.png' alt='Home'/> 
	  		</NavLink>
		</div>

		<button className="header__nav-button" onClick={() => {
	  		dispatch(toggleNav())
		}}>
	  		{!nav ? '☰':'✕'}
		</button>    
  	</header>
) 

const mapStateToProps = (state) => {
	return {
		nav: state.navigation.nav
  	}
}

export default connect(mapStateToProps)(Header)