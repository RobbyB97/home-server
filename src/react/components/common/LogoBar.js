import React from 'react'

const LogoBar = () => (
    <div id="logo-bar">
        <ul>
            <li>
                <a href="https://www.github.com/robbyb97" target="_blank">
                    <img src="/dist/images/icons/github.svg" />
                </a>
            </li>
            
            <li>
                <a href="https://www.linkedin.com/in/robbyb97" target="_blank">
                    <img src="/dist/images/icons/linkedin.png" />
                </a>
            </li>
        </ul>
    </div>
)

export default LogoBar