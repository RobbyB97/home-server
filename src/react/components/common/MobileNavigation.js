import React from 'react'
import {NavLink} from 'react-router-dom'
import { connect } from 'react-redux'

import {toggleNav, toggleNavOff} from '../../actions/navigation'
import {setTitle} from '../../actions/title'
import {homeOn, homeOff} from '../../actions/home'


const MobileNavigation = ({nav, dispatch, home}) => (
    <div id="mobile-navigation" className={nav ? 'active-nav':''}>
        <div className="mobile-navigation__background">
            <ul className="mobile-navigation__intra-nav">
                <li>
                    <NavLink to="/" onClick={() => {
                        dispatch(toggleNav())
                        dispatch(setTitle(''))
                        dispatch(homeOn())
                    }}>Home</NavLink>
                </li>

                <li>
                    <NavLink to="/skills/" onClick={() => {
                        dispatch(toggleNav())
                        dispatch(setTitle('Skills'))
                        dispatch(homeOff())
                    }}>Skills</NavLink>
                </li>

                <li>
                    <NavLink to="/experience/" onClick={() => {
                        dispatch(toggleNav())
                        dispatch(setTitle('Experience'))
                        dispatch(homeOff())
                    }}>Experience</NavLink>
                </li>

                <li>
                    <NavLink to="/projects/" onClick={() => {
                        dispatch(toggleNav())
                        dispatch(setTitle('Projects'))
                        dispatch(homeOff())
                    }}>Projects</NavLink>
                </li>

                <li>
                    <NavLink to="/contact/" onClick={() => {
                        dispatch(toggleNav())
                        dispatch(setTitle('Contact'))
                        dispatch(homeOff())
                    }}>Contact</NavLink>
                </li>
            </ul>
        </div>
    </div>
)

const mapStateToProps = (state) => {
    return {
        nav: state.navigation.nav,
        home: state.home.home
    }
}

export default connect(mapStateToProps)(MobileNavigation)