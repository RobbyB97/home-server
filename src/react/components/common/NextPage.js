import React from 'react'
import {NavLink} from 'react-router-dom'


export class NextPage extends React.Component {
    

    render() {
        return (
            <div className={this.props.float ? "nextPage float" : "nextPage"}>
                <NavLink 
                    to={this.props.route ? this.props.route : '/'}
                >
                    <p className="nextPage__text">
                        {this.props.text && this.props.text} â†’
                    </p>
                </NavLink>
            </div>
        )
    }
}


export default NextPage