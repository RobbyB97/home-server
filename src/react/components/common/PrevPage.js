import React from 'react'
import {NavLink} from 'react-router-dom'


export class PrevPage extends React.Component {
    

    render() {
        return (
            <div className="prevpage">
                {this.props.text &&
                    <p className="prevpage__text">
                        {this.props.text}
                    </p>
                }
                <NavLink 
                    to={this.props.route ? this.props.route : '/'}
                >
                    <div className="prevpage__arrow">
                        <img src="/dist/images/icons/down_arrow_white.png"/>
                    </div>
                </NavLink>
            </div>
        )
    }
}


export default PrevPage