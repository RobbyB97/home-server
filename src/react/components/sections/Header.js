import React from 'react';

import DesktopNavigation from '../ui/DesktopNavigation';
import MobileNavigation from '../ui/MobileNavigation';
import HomeButton from '../ui/HomeButton';



export class Header extends React.Component {
    render() {
        return (
            <header id="Header" className="header">
                <HomeButton internal />
                <DesktopNavigation />
                <MobileNavigation />
            </header>
        );
    }
}


export default Header;