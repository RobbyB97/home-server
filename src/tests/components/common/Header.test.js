import React from 'react'
import {shallow} from 'enzyme'

import {Header} from '../../../react/components/common/Header'


let wrapper

beforeEach(() => {
    wrapper = shallow(<Header />)
})


test('Render Header', () => {
    expect(wrapper).toMatchSnapshot()
})