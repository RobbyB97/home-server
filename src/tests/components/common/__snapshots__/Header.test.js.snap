// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header 1`] = `
<header
  id="header"
>
  <div
    className="header__logo"
  >
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      onClick={[Function]}
      to="/"
    >
      <img
        alt="Home"
        src="/dist/images/favicon.png"
      />
    </NavLink>
  </div>
  <button
    className="header__nav-button"
    onClick={[Function]}
  >
    â˜°
  </button>
</header>
`;
